packages/connect-form/src/utils/validation.ts

```diff
@@ -21,10 +23,12 @@ export type FieldName =
 export type ConnectionFormError = {
```

Comment:
would be better to change this type like this:

```
type ConnectionFormErrorWithField = {fieldName: FieldName, fieldTab: FieldTab, fieldIndex?: number, message: string};
type ConnectionFormErrorWithoutField = {message: string};
ConnectionFormError = ConnectionFormErrorWithField | ConnectionFormErrorWithoutField;
```

so we won't ever forget to add the tab if the error relates to a field

Reply:
If you defined it like that, you will be able to define these `field*` properties to anything that has a `ConnectionFormError`, but not access them because the union type only have `message` property in common. I think what you're looking for here is this:

```ts
type ConnectionFormErrorWithField =  { fieldName: FieldName, fieldTab: FieldTab, fieldIndex?: number, message: string };
type ConnectionFormErrorWithoutField = { message: string };
type ConnectionFormError = ConnectionFormErrorWithField | ConnectionFormErrorWithoutField;
```

This explicitly tells TS that these properties will never occur on the object and allows typescript to pattern match on keys similar to how it can do on constant values. [Here's a playground](https://www.typescriptlang.org/play?#code/LAKALgngDgpgBAYQPYDsUwMZgJaoGJIBOAtgKKGFEDq2YAFntjADYAmcAvHHAN5wBmTNgDkAhsRgAuOAGcwhbCgDmAGgFDWAFVEAjaXIXK1glqwCSKVjAAeAfmkoArsR0xCaiTJmilU2fMUlOABfAG5QSFhEVHQsXBQCEnJKQhp6JEcwRlNOXjhPb199AOUQ8JAAegq4SPhkNEwcfCIyCmpaOgysjVyeExFxGHs4dAA3NzVudVNtHWGxian+80sbeZhx924Cnz8DQLCI6DqYxvjE1pTc+timhJbk9vpstjgAH2iGuOakttSOrovVjlUCjUSEOCiaQ3M4-S5EXr5GBeXbSADk-CQSDRxg0Ygk6J04LRZVAoH4jhQ3xQcF83VM+JgAAo3CloadqRdHoQAJS8UDcQgwMCOQg01lEAB0y0Z5WCQA) if you want to try it out yourself

Reply:
Not all tab errors are assigned to fields:
```js
{
  fieldTab: 'TLS/SSL',
  message: 'TLS must be enabled in order to use x509 authentication.',
}
```
Should this one have the tls fieldName maybe? I think we'll have to use it in the ui then. I like the idea of this type being as strict as possible, but if we'll need to update those then.

Reply:
> `type ConnectionFormErrorWithoutField = { fieldName?: never, fieldTab?: never, fieldIndex?: never, message: string };`

@gribnoysup that's super neat!

Reply:
Updated the types - also found a spot or two we were missing a fieldTab after adding it! Types are so nice. Thanks for the good suggestion.